modules/core/src/utils/tesselator.js

```diff
@@ -57,6 +57,7 @@ export default class Tesselator {
     this.getGeometry = getGeometry;
     this.positionSize =
       (geometryBuffer && geometryBuffer.size) || (positionFormat === 'XY' ? 2 : 3);
+    assert(this.positionSize > 1 && this.positionSize < 4);
```

Comment:
In what circumstances could this be untrue? Avoid assertions that may break the app but do not provide any debugging information to the user.

Reply:
While implementing this feature I came across a situation where this.positionSize was somehow -Infinity which led to infinite loops and out of memory crashes which were a pain to debug as the browser would freeze. It is most likely this was due to a bug in the loader I was modifying so you could argue that this assertion is unneeded. On the other hand, were it there it would have saved me a lot of time.

Given that the code assumes this value to be 2 or 3 I felt that having an assertion there to be safe and to validate the input wouldn't hurt. However if you disagree, I'm happy to remove it.

Reply:
Yes this should be fixed in the loader.

The criteria for adding assertions is whether they provide actionable items for the user to fix errors in their own code. You may add these during development but they should not be shipped to the end user.

Reply:
Fixed